### ISSUE TITLE
Godot 4.3 Resource icons!

### ISSUE THREAD

**Issue by firedfromlife** [dialogic-godot/dialogic#2355]
**Is your feature request related to a problem? Please describe.**
**Finally**, as of Godot 4.3 custom resources can have their own icons! Timelines and Characters should absolutely support this

**Describe the solution you'd like**
Add the new @ icon annotation to those classes.

**Additional context**
I intend to submit a PR for this, though it seems @ icon annotation doesn't work correctly for custom extensions?

**Comment by firedfromlife** [comment]
I also realized, there currently isn't a custom icon for the timelines. It just uses the three line hamburger menu icon. I came up with one real quick, any thoughts?

I fear it may not read at lower resolutions so I may change it.

![image](https://github.com/user-attachments/assets/419e3587-92a9-441e-a5d1-311258dddfa1)

**Comment by Jowan-Spooner** [comment]
Yeah, I think this won't work as an icon. Feel free to try out some more though! We certainly need an svg. You can find the svgs of all the godot icons here, in case it helps 😋: https://github.com/godotengine/godot/tree/master/editor/icons

I think timeline is a hard concept to put into a small icon, so I assume we can't get much better then what we have right now, but maybe there is a way to make it NOT look like a hamburger menu :)?
Good luck, I'll test the sidebar PR tomorrow, for now I'm of to bed...

**Comment by firedfromlife** [comment]
**A couple other options:**
- Simple
    - ![-Asset 1](https://github.com/user-attachments/assets/bfb603bc-f5e4-4f2c-b405-88e935275754)
- More speech bubble like
    - ![-Asset 2](https://github.com/user-attachments/assets/c9384c61-08b5-4735-a305-cc2798f5063d)

Am done for today, but let me know.

**Comment by firedfromlife** [comment]
I  filed an issue with with Godot engine, and the issue has now been confirmed in 4.3 rc1. Custom icons for resources do not work with custom file extensions at the moment... At least in the file system. Hopefully will get fixed prior to 4.3 release.
